### ISSUE TITLE
[KubeRay][Dashboard] Port-forwarding dashboard fails. 

### ISSUE THREAD

**Issue by DmitriGekhtman** [ray-project/ray#25909]
### What happened + What you expected to happen

Port-forwarding the Ray dashboard fails with KubeRay in at least some Kubernetes setups.
I've observed the problem on GKE but not on a local KinD cluster.

Start a KubeRay Ray Cluster and attempt to port-forward the Ray head's dashboard port
`kubectl -n ray port-forward service/example-cluster-ray-head 8265:8265`
```
Forwarding from 127.0.0.1:8265 -> 8265
Forwarding from [::1]:8265 -> 8265
Handling connection for 8265
E0317 22:21:58.501309  135973 portforward.go:406] an error occurred forwarding 8265 -> 8265: error forwarding port 8265 to pod 755dce9c462676627f07602de97f7bf9c52ab727336cda0d7d02f0b566c5d292, uid : failed to execute portforward in network namespace "/var/run/netns/cni-76ec8b4c-cc4c-49cb-aa89-f87ec7f5f9e4": failed to dial 8265: dial tcp4 127.0.0.1:8265: connect: connection refused
E0317 22:21:58.501671  135973 portforward.go:234] lost connection to pod
```

Context: https://discuss.ray.io/t/ray-job-submit-errors-on-kubernetes/5449/4

### Versions / Dependencies

Ray master, KubeRay master, GKE and potentially other cloud providers.

### Reproduction script

Start a KubeRay Ray Cluster and attempt to port-forward the Ray head's dashboard port
`kubectl -n ray port-forward service/example-cluster-ray-head 8265:8265`

### Issue Severity

High: It blocks me from completing my task.

**Comment by antoinekrajnc** [comment]
I have the same issue. @DmitriGekhtman did you happen to find a solution ?

**Comment by DmitriGekhtman** [comment]
I haven't gotten a chance to look into it yet -- but will very soon.
The current workaround is to submit jobs from within your Kubernetes cluster.

@antoinekrajnc what kind of K8s environment do you use?

**Comment by DmitriGekhtman** [comment]
Ok, starting the investigation:
I did not hit the issue with Ray 1.11.0, which is good news. This means I can start bisecting to identify the breaking commit.

**Comment by DmitriGekhtman** [comment]
...failed to reproduce on Ray 1.13.0.
Going to keep hunting for the reproduction...

**Comment by DmitriGekhtman** [comment]
I'm unfortunately unable to reproduce this with any Ray version at the moment.
Maybe this is dependent on the internet connection of the laptop that is trying to port-forward?

That would be consistent with me running into issues from home on GKE but not from the office with GKE, and also not on minikube.

I guess I'll give this another shot from somewhere with a bad wifi connection.

**Comment by DmitriGekhtman** [comment]
I’d be glad to schedule a time to pair-debug with anyone experiencing the port-forwarding issue!

**Comment by DmitriGekhtman** [comment]
Going to re-open as there's some indication that the issue is still occurring for some users.
https://github.com/ray-project/ray/issues/25909

**Comment by DmitriGekhtman** [comment]
^ Apparently, allowing istio to inject sidecars into your KubeRay pods can trigger this issue too.

**Comment by DmitriGekhtman** [comment]
I'm just going to leave this open as low-priority, so we can accumulate more info.

**Comment by chrisk314** [comment]
I'm seeing this issue in an AKS cluster. Dashboard was initially working fine for a couple of days. Coming back a couple weeks later we're now seeing this.

Kubernetes version 1.23.5
Ray version 1.13.0
Python version 3.7.7
Localhost OS: Ubuntu 20.04

**Comment by DmitriGekhtman** [comment]
@chrisk314 do you know if there are any other details about the K8s environment that might be relevant?

For one user, the issue had to do with istio.

Another question -- are you able to port-forward the Ray Client port 10001?

Finally, there should be a service configured for the RayCluster. Could you share the output of `kubectl get service <name of service>`?

**Comment by chrisk314** [comment]
Hmm, not sure really. We are using the k8s cluster for a few different things but we don't have anything that should impact networking in a significant way like an ingress or istio etc. I can tell you that we deployed our Ray cluster by manually cloning the Ray Github repo as per the docs, and running Helm install, rather than using KubeRay. I can also tell you we deployed in a namespace.

Unfortunately we uninstalled and reinstalled the Ray cluster so I can't show you the outputs from the problematic Ray cluster. The good news is that this fixed the dashboard. However, I did look at the Ray Head Service earlier and can confirm the details were the same as below. Also, the Ray client was working without issue, only the dashboard was having issues.

Here's what the current - well behaved - Ray Head Service looks like
```
$ kubectl get svc -n ray ray-cluster-ray-head
NAME                   TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)                       AGE
ray-cluster-ray-head   ClusterIP   10.6.146.15   <none>        10001/TCP,8265/TCP,8000/TCP   77m
```

Is there a server process for the dashboard running somewhere which maybe dies after some time?

**Comment by DmitriGekhtman** [comment]
Thanks for the update!

> Is there a server process for the dashboard running somewhere which maybe dies after some time?

There is indeed a dashboard server process and it is possible there was an issue with it.
By the way, dashboard logs are written to `/tmp/ray/session_latest/logs/dashboard.log` in the head pod.

cc @brucez-anyscale @rkooo567 @alanwguo for general awareness

**Comment by rkooo567** [comment]
@chrisk314 were you able to fix the issue? Let us know if you find the root cause.

**Comment by chrisk314** [comment]
@rkooo567 I don't have anything to report right now. I wasn't using Ray much over the last few months. Just picking it up again since last week but on v2.0.1 now. Haven't seen this problem so far but also haven't been using the dashboard heavily. Will report back with details from logs if I hit this again.

**Comment by DmitriGekhtman** [comment]
If anyone encounters this again and would like to continue the discussion, please open a [bug report in the KubeRay repo](https://github.com/ray-project/kuberay/issues/new?assignees=&labels=bug%2Ctriage&template=bug-report.yml&title=%5BBug%5D+).

**Comment by will-hang** [comment]
> The solution appears to be to install Ray with `pip install ray[default]` locally! (This command installs the dashboard dependencies, while `pip install ray` does not.) https://docs.ray.io/en/latest/ray-overview/installation.html
> 
> We'll update the documentation accordingly.
> 
> Please re-open if the issue is encountered again with this recommendation.

Yep, this helped us solve the issue on our k8s deployment. We were only including the link to the whl instead of the `ray[default] @` prefix.
